### ISSUE TITLE
Update Python version on Windows installer to 3.10 to not flag `match` as an error

### ISSUE THREAD

**Issue by acolov** [spyder-ide/spyder#19502]
### What steps will reproduce the problem?

I've been using python 3.10 on spyder 5.3.3, using the external interpreter to be able to use the last python version (standard included spyder python is 3.8.10). Everything works fine but the pyflakes linter gives sintax errors whenever I use the match - case structure. 

This is pretty annoying and I tried to install pyflakes 2.5.0 (which should in theory solve the issue) but there is no effect on Spyder. I installed it from my external terminal, thus I thought that Spyder was still trying to get the internal version of spyder. It turns out that the internal version of pyflakes was still 2.4.0, and I couldn't upgrade it to 2.5.0 beacause it is not supported by python 3.8.10.

Is it possible to have pyflakes read from my external python build?

Thanks

## Versions

* Spyder version: 5.3.3  (standalone)
* Python version: 3.8.10 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                 :  1.4.1 (OK)
chardet >=2.0.0                      :  5.0.0 (OK)
cloudpickle >=0.5.0                  :  2.1.0 (OK)
cookiecutter >=1.6.0                 :  2.1.1 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree                         :  None (OK)
IPython >=7.31.1;<8.0.0              :  7.34.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.1 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.14.0 (OK)
keyring >=17.0.0                     :  23.8.2 (OK)
nbconvert >=4.0                      :  7.0.0 (OK)
numpydoc >=0.6.0                     :  1.4.0 (OK)
paramiko >=2.4.0                     :  2.11.0 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.1 (OK)
pygments >=2.0                       :  2.13.0 (OK)
pylint >=2.5.0;<3.0                  :  2.15.0 (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0                 :  1.5.0 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.1.10                   :  0.2.1 (OK)
qtawesome >=1.0.2                    :  1.1.1 (OK)
qtconsole >=5.3.2;<5.4.0             :  5.3.2 (OK)
qtpy >=2.1.0                         :  2.2.0 (OK)
rtree >=0.9.7                        :  1.0.0 (OK)
setuptools >=49.6.0                  :  65.3.0 (OK)
sphinx >=0.6.6                       :  5.1.1 (OK)
spyder_kernels >=2.3.3;<2.4.0        :  2.3.3 (OK)
textdistance >=4.2.0                 :  4.4.0 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog                             :  2.1.9 (OK)
zmq >=22.1.0                         :  23.2.1 (OK)

# Optional:
cython >=0.21                        :  0.29.32 (OK)
matplotlib >=3.0.0                   :  3.5.3 (OK)
numpy >=1.7                          :  1.22.4 (OK)
pandas >=1.1.1                       :  1.4.3 (OK)
scipy >=0.17.0                       :  1.9.1 (OK)
sympy >=0.7.3                        :  1.11 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)
```

**Comment by ccordoba12** [comment]
Hey @acolov, thanks for reporting. We're going to increase the Pyflakes version we use in the next Spyder version (5.4.0), to be released next month.

**Comment by ccordoba12** [comment]
Sorry, I just saw you said

> Is it possible to have pyflakes read from my external python build?

Sorry, that's not possible. You have to wait until 5.4.0 is out.

**Comment by acolov** [comment]
Alright thank you anyway. I'll wait for the 5.4.0 then ;)

**Comment by ccordoba12** [comment]
Just in case, @dalthviz, could you check if we're still showing errors in the editor for `match` cases using our Windows installer built in our latest pull requests?

**Comment by dalthviz** [comment]
@ccordoba12 I think the installers build on PRs for Windows uses `python-lsp-server`1.5.0 (`python-lsp-server` is not being installed from the external-deps folder, only `spyder-kernels` and `qdarkstyle`). Also, just in case, the way Spyder will depend on `Pyflakes` >=2.5.0 will be due to an update to constraint `python-lsp-server` to be >=1.6.0 for Spyder 5.4.0 (which will include these changes: https://github.com/python-lsp/python-lsp-server/pull/257).

So the installers for Windows (even the ones that come from PRs) use for the moment `Pyflakes` 2.4.0 and `python-lsp-server` 1.5.0

**Comment by ccordoba12** [comment]
Ok, I tested this locally with Spyder running on Python 3.7 and Pyflakes 2.5.0, and Pyflakes still marks `match` as an error. So I think the only solution for this is to update the Python version that comes with our Windows installer to 3.10.

However, that's going to take us between three to six months because it's not that easy.

**Comment by realitychemist** [comment]
Just curious about the status of this update.  I'm running python `3.10.9` externally, and I'm on the Windows version `5.4.2` of Spyder which still seems to be using python version `3.8.10` internally.  Of course the code still runs even when flagged up as being in error, so it's not crazy urgent or anything, but it would be nice not to have spurious syntax errors flagged up in my scripts.

**Comment by ccordoba12** [comment]
This will be fixed in Spyder 6.0, to be released in the second half of the year.

**Comment by ccordoba12** [comment]
Closing because our new Windows, Mac and Linux installers, which are going to be based on Conda, will come with a more recent Python version.
